File: Documentation/snippets/fibonacci_sequence.py
def generate_fibonacci_sequence(n: int) -> list:
    """Generates the Fibonacci sequence up to the n-th number.

    Args:
        n: The length of the Fibonacci sequence.

    Returns:
        A list containing the Fibonacci sequence up to the n-th number.
    """
    if n <= 0:
        return []
    sequence = [0, 1]
    while len(sequence) < n:
        sequence.append(sequence[-1] + sequence[-2])
    return sequence[:n]

# Example usage:
n = 10
print(f"The Fibonacci sequence up to the {n}-th number is: {generate_fibonacci_sequence(n)}")

File: Documentation/snippets/cpp/fibonacci_sequence.cpp
#include <iostream>
#include <vector>

std::vector<int> generate_fibonacci_sequence(int n) {
    std::vector<int> sequence;
    if (n >= 1) {
        sequence.push_back(0);
    }
    if (n >= 2) {
        sequence.push_back(1);
    }

    for (int i = 2; i < n; ++i) {
        sequence.push_back(sequence[i - 1] + sequence[i - 2]);
    }

    return sequence;
}

// Example usage:
int main() {
    int n = 10;
    std::vector<int> sequence = generate_fibonacci_sequence(n);
    std::cout << "The Fibonacci sequence up to the " << n << "-th number is: ";
    for (int num : sequence) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    return 0;
}
